src/hotspot/share/opto/c2_globals.hpp

```diff
@@ -470,6 +470,9 @@
   notproduct(bool, TraceReduceAllocationMerges, false,                      \
           "Trace decision for simplifying allocation merges.")              \
                                                                             \
+  notproduct(bool, VerifyReduceAllocationMerges , true,                     \
```

Comment:
From changes I assume you want verification done in debug VM. Use `develop(`  flag declaration for that.
(`notproduct` includes `optimized` VM build).